intrinsic_cal/src/rail_cal.cpp

```diff
@@ -113,22 +119,23 @@ RailCalService::RailCalService(ros::NodeHandle nh)
   if(!pnh.getParam( "camera_spacing", camera_spacing_)){
     ROS_ERROR("Must set param:  camera_spacing");
   }
-  if(!pnh.getParam( "image_height", image_height_)){
-    ROS_ERROR("Must set param:  image_height_");
-  }
-  if(!pnh.getParam( "image_width", image_width_)){
-    ROS_ERROR("Must set param:  image_width_");
-  }
   if(!pnh.getParam( "target_to_rail_distance", D0_)){
     ROS_ERROR("Must set param:  target_to_rail_distance");
   }
 
+  u_int32_t queue_size = 5;
+  rgb_sub_ = nh_.subscribe("color_image", queue_size, &RailCalService::cameraCallback, this);
+  rgb_pub_ = nh_.advertise<sensor_msgs::Image>("color_image_center", 1);
+
   // setting camera extrinsic parameters in no longer necessary they are not used
-  tf::Matrix3x3 R; 
+  tf::Matrix3x3 R;
   Pose6d pose;
-  R[0][0] = 0;   R[0][1] = 0;   R[0][2] = -1;
-  R[1][0] = 1;   R[1][1] = 0;   R[1][2] = 0;
-  R[2][0] = 0;   R[2][1] = -1;   R[2][2] = 0;
+  //R[0][0] = 0;   R[0][1] = 0;   R[0][2] = -1;
```

Comment:
This is the camera position.  So for an image, the x-axis is pointing to the right, the y-axis is point downward, which means that the z-axis is pointing toward the target.